apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -766,6 +766,30 @@ defmodule Astarte.AppEngine.API.Device do
     end
   end
 
+  defp cast_value(:longinteger, string_value) when is_binary(string_value) do
+    case Integer.parse(string_value) do
+      {int_value, ""} ->
+        {:ok, int_value}
+
+      _ ->
+        {:error, :unexpected_value_type, expected: :longinteger}
+    end
+  end
+
+  defp cast_value(:longinteger, _string_value) do
+    {:error, :unexpected_value_type, expected: :longinteger}
+  end
+
+  defp cast_value(:longintegerarray, string_values) do
```

Comment:
Nit: you don't know yet if values are string or something else, so just call the variable `values`